File: public/lib/fidl/compiler/src/fidl/compiler/generators/go/translator/test_util.go
// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package translator

import (
	"testing"
)

func checkEq(t *testing.T, expected, actual interface{}) {
	if expected != actual {
		errorMsg := "Failed check: Expected (%q), Actual (%q)"
		if _, ok := expected.(string); !ok {
			errorMsg = "Failed check: Expected (%s), Actual (%s)"
		}
		t.Fatalf(errorMsg, expected, actual)
	}
}

File: garnet/public/lib/component/cpp/testing/test_util.cc
// Copyright 2018 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "lib/component/cpp/testing/test_util.h"

#include <lib/fdio/limits.h>
#include <lib/fdio/util.h>

namespace component {
namespace testing {

zx::channel OpenAsDirectory(fs::Vfs* vfs, fbl::RefPtr<fs::Vnode> node) {
  zx::channel h1, h2;
  if (zx::channel::create(0, &h1, &h2) != ZX_OK)
    return zx::channel();
  if (vfs->ServeDirectory(std::move(node), std::move(h1)) != ZX_OK)
    return zx::channel();
  return h2;
}

}  // namespace testing
}  // namespace component
